ngraph/test/type_prop/negative.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/type_prop.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(type_prop, negative_inference_shape)
+{
+    auto param = std::make_shared<op::Parameter>(element::i32, Shape{21, 15, 2});
+    auto op = std::make_shared<op::Negative>(param);
+    ASSERT_EQ(op->get_shape(), (Shape{21, 15, 2}));
+}
+
+TEST(type_prop, negative_input_type_inference_i32)
+{
+    auto param = std::make_shared<op::Parameter>(element::i32, Shape{21, 15, 2});
+    auto op = std::make_shared<op::Negative>(param);
+    ASSERT_EQ(op->get_element_type(), element::i32);
+}
+
+TEST(type_prop, negative_input_type_inference_i64)
+{
+    auto param = std::make_shared<op::Parameter>(element::i64, Shape{21, 15, 2});
+    auto op = std::make_shared<op::Negative>(param);
+    ASSERT_EQ(op->get_element_type(), element::i64);
+}
+
+TEST(type_prop, negative_input_type_inference_f32)
+{
+    auto param = std::make_shared<op::Parameter>(element::f32, Shape{21, 15, 2});
+    auto op = std::make_shared<op::Negative>(param);
+    ASSERT_EQ(op->get_shape(), (Shape{21, 15, 2}));
+}
+
+TEST(type_prop, negative_input_type_inference_f64)
+{
+    auto param = std::make_shared<op::Parameter>(element::f32, Shape{21, 15, 2});
+    auto op = std::make_shared<op::Negative>(param);
+    ASSERT_EQ(op->get_shape(), (Shape{21, 15, 2}));
+}
+
+TEST(type_prop, dynamic_rank_input_shape_2D)
+{
+    const PartialShape param_shape{Dimension::dynamic(), 10};
+    const auto param = std::make_shared<op::Parameter>(element::f32, param_shape);
+    const auto op = std::make_shared<op::Negative>(param);
+    ASSERT_TRUE(op->get_output_partial_shape(0).same_scheme(PartialShape{Dimension(), 10}));
+}
+
+TEST(type_prop, dynamic_rank_input_shape_3D)
+{
+    const PartialShape param_shape{100, Dimension::dynamic(), 58};
+    const auto param = std::make_shared<op::Parameter>(element::f32, param_shape);
+    const auto op = std::make_shared<op::Negative>(param);
+    ASSERT_TRUE(op->get_output_partial_shape(0).same_scheme(PartialShape{100, Dimension(), 58}));
+}
+
+TEST(type_prop, dynamic_rank_input_shape_full)
+{
+    const auto param = std::make_shared<op::Parameter>(element::f64, PartialShape::dynamic());
+    const auto op = std::make_shared<op::Negative>(param);
+    ASSERT_TRUE(op->get_output_partial_shape(0).same_scheme(PartialShape::dynamic()));
+}
```

Comment:
Can we use typed test to significantly reduce the size of the code?

Reply:
Done